src/Acuminator/Acuminator.Analyzers/Analyzers/DAC/DacPropertyGraphUsageAnalyzer.cs

```diff
@@ -0,0 +1,135 @@
+
+            if (property != null && IsDacProperty(property, pxContext))
+            {
+                IEnumerable<MemberAccessExpressionSyntax> membersAccess = node.DescendantNodes().OfType<MemberAccessExpressionSyntax>();
+
+                foreach (MemberAccessExpressionSyntax ma in membersAccess)
+                {
+                    AnalyzeMemberAccessName(ma.Name, syntaxContext, pxContext);
+                    AnalyzeMemberAccessExpression(ma.Expression, syntaxContext, pxContext);
+                }
+
+                IEnumerable<LocalDeclarationStatementSyntax> localVariables = node.DescendantNodes().OfType<LocalDeclarationStatementSyntax>();
+
+                foreach (LocalDeclarationStatementSyntax l in localVariables)
+                {
+                    AnalyzeLocalDeclaration(l, syntaxContext, pxContext);
+                }
+            }
+        }
+
+        private void AnalyzeLocalDeclaration(LocalDeclarationStatementSyntax local, SyntaxNodeAnalysisContext syntaxContext, PXContext pxContext)
+        {
+            bool isGraphDeclarationType = syntaxContext.SemanticModel.GetSymbolInfo(local.Declaration.Type).Symbol is ITypeSymbol typeSymbol && typeSymbol.InheritsFromOrEquals(pxContext.PXGraphType);
+
+            if (isGraphDeclarationType)
+            {
+                syntaxContext.ReportDiagnostic(Diagnostic.Create(Descriptors.PX1029_PXGraphUsageInDacProperty, local.GetLocation()));
+            }
+        }
+
+        private void AnalyzeMemberAccessExpression(ExpressionSyntax expression, SyntaxNodeAnalysisContext syntaxContext, PXContext pxContext)
+        {
+            ISymbol expressionSymbol = syntaxContext.SemanticModel.GetSymbolInfo(expression, syntaxContext.CancellationToken).Symbol;
+
+            if (expressionSymbol != null && IsGraphUsedAsMemberExpression(expressionSymbol, pxContext))
+            {
+                syntaxContext.ReportDiagnostic(Diagnostic.Create(Descriptors.PX1029_PXGraphUsageInDacProperty, expression.GetLocation()));
+            }
+        }
+
+        private void AnalyzeMemberAccessName(SimpleNameSyntax name, SyntaxNodeAnalysisContext syntaxContext, PXContext pxContext)
+        {
+            ISymbol nameSymbol = syntaxContext.SemanticModel.GetSymbolInfo(name, syntaxContext.CancellationToken).Symbol;
+
+            if (nameSymbol != null && IsGraphUsedAsMemberName(nameSymbol, pxContext))
+            {
+                syntaxContext.ReportDiagnostic(Diagnostic.Create(Descriptors.PX1029_PXGraphUsageInDacProperty, name.GetLocation()));
+            }
+        }
+
+        private bool IsGraphUsedAsMemberExpression(ISymbol expressionSymbol, PXContext pxContext)
+        {
+            bool isGraphUsed = false;
+
+            switch (expressionSymbol)
+            {
+                case ILocalSymbol l:
+                    if (l.Type.InheritsFromOrEquals(pxContext.PXGraphType))
```

Comment:
@metallmonster , why don't you use IsPXGraph extension helper method? 
If you don't use it because IsPXGraph performs check by only by names, then please add overload to utils which accepts PXContext and checks by comparing with PXGraphType object. We should keep general checks like check for graph or extension in helpers in order to keep them consistent and re-use them